app/pages/reservation/reservation-information/InternalReservationForm.js

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { Component } from 'react';
+import { Field, reduxForm } from 'redux-form';
+import Form from 'react-bootstrap/lib/Form';
+import Row from 'react-bootstrap/lib/Row';
+import Col from 'react-bootstrap/lib/Col';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import injectT from '../../../i18n/injectT';
+import FormTypes from '../../../constants/FormTypes';
+
+export const maxLengths = {
+  internalReservationComments: 1500
+};
+
+function mapStateToProps(state) {
+  return { stateForm: state.form };
+}
+
+class UnconnectedInternalReservationForm extends Component {
+  state = {
+    internalReservationDefaultChecked: true,
+    markAsClosedDefaultChecked: false
+  }
+
+  componentDidMount() {
+    this.props.dispatch(this.internalReservationFormInitialValue());
+  }
+
+  internalReservationFormInitialValue() {
+    return {
+      type: '@@redux-form/CHANGE',
+      meta: {
+        form: 'RESERVATION',
+        field: 'internalReservation',
+        touch: false,
+        persistentSubmitErrors: false
+      },
+      payload: this.state.internalReservationDefaultChecked
+    };
+  }
+
+  render() {
+    const {
+      t,
+      stateForm
+    } = this.props;
+    let internalReservationCommentsLength = 0;
+    if (
+      stateForm
+      && stateForm.RESERVATION
+      && stateForm.RESERVATION.values
+      && stateForm.RESERVATION.values.internalReservationComments
+    ) internalReservationCommentsLength = stateForm.RESERVATION.values.internalReservationComments.length;
+    return (
+      <div className="app-ReservationDetails">
+        <Form className="reservation-form" horizontal noValidate>
+          <h2 className="app-ReservationPage__title">{t('ReservationForm.premiseStaffOnly')}</h2>
+          <Row>
+            <Col md={1}>
+              <label className="custom-checkbox-container">
+                <Field
+                  checked={this.state.internalReservationDefaultChecked}
+                  component="input"
+                  id="internalReservationChecked"
+                  name="internalReservation"
+                  onChange={() => this.setState(
+                    prevState => ({ internalReservationDefaultChecked: !prevState.internalReservationDefaultChecked })
+                  )}
+                  type="checkbox"
+                />
+                <span className="custom-checkmark" />
+              </label>
+            </Col>
+            <Col md={11}>
+              <span className="app-ReservationDetails__value">
+                {t('ReservationForm.internalReservation')}
+                <br />
+                {t('ReservationForm.internalReservationDescription')}
+              </span>
+            </Col>
+          </Row>
+          <Row>
+            <Col md={1}>
+              <label className="custom-checkbox-container">
```

Comment:
if this never get reused in the future, give it BEM className, not normal className like this

Reply:
Changed to `app-InternalReservationForm__checkbox`